pkg/dao/rpm_test.go

```diff
@@ -241,14 +253,94 @@ func (s *RpmSuite) TestRpmSearch() {
 	}
 }
 
+// func (s *RpmSuite) randomPackageName(size int) string {
+func randomPackageName(size int) string {
+	const lookup string = "0123456789abcdefghijklmnopqrstuvwxyz"
+	return seeds.RandStringWithTable(size, lookup)
+}
+
+// func (s *RpmSuite) randomHexadecimal(size int) string {
+func randomHexadecimal(size int) string {
+	const lookup string = "0123456789abcdef"
+	return seeds.RandStringWithTable(size, lookup)
+}
+
+// func (s *RpmSuite) randomYumPackage() yum.Package {
+func randomYumPackage(pkg *yum.Package) {
+	if pkg == nil {
+		return
+	}
+	pkgName := randomPackageName(32)
+	pkg.Name = pkgName
+	pkg.Arch = "x86_64"
+	pkg.Summary = pkgName + " summary"
+	pkg.Version = yum.Version{
+		Version: "1.0.0",
+		Release: "dev",
+		Epoch:   0,
+	}
+	pkg.Type = "rpm"
+	pkg.Checksum = yum.Checksum{
+		Type:  "sha256",
+		Value: randomHexadecimal(64),
+	}
+}
+
+func makeYumPackage(size int) []yum.Package {
+	var pkgs []yum.Package = []yum.Package{}
+
+	if size <= 0 {
+		return pkgs
+	}
+
+	pkgs = make([]yum.Package, size)
```

Comment:
```suggestion
	pkgs := make([]yum.Package, size)
	if size <= 0 {
		return pkgs
	}
```

Reply:
i guess this wouldn't cover a negative number being passed in, but its test code

Reply:
thanks! I have updated; now it launch a panic when a negative number is provided, and for a size equals to 0 the empty slice.